hyp3lib/get_orb.py

```diff
@@ -1,197 +1,140 @@
 """Get Sentinel-1 orbit file(s) from ASF or ESA website"""
 
-from __future__ import print_function, absolute_import, division, unicode_literals
-
-import re
-from lxml import html
+import argparse
+import logging
 import os
+import re
+import sys
 from datetime import datetime, timedelta
-import argparse
-from hyp3lib.verify_opod import verify_opod
+
 import requests
-import json
+from lxml import html
 from requests.adapters import HTTPAdapter
 from six.moves.urllib.parse import urlparse
 
+from hyp3lib import OrbitDownloadError
+from hyp3lib.fetch import download_file
+from hyp3lib.verify_opod import verify_opod
 
-class FileException(Exception):
-  """Could not download orbit file"""
 
+def _get_asf_orbit_url(search_url, platform, timestamp, verify=True):
+    if not search_url.endswith('/'):
+        search_url += '/'
 
-def getPageContentsESA(url, verify):
-    print("Getting result of {}".format(url))
-    hostname = urlparse(url).hostname
+    hostname = urlparse(search_url).hostname
     session = requests.Session()
     session.mount(hostname, HTTPAdapter(max_retries=10))
-    page = session.get(url, timeout=60, verify=verify)
-    print(page)
-    results = json.loads(page.text)
-    print("results : {}".format(results))
-    if results['count'] > 0:
-        print(results['results'][0]['physical_name'])
-        fileName = results['results'][0]['physical_name']
-        url = results['results'][0]['remote_url']
-        return(fileName,url)
-    else:
-        print("WARNING: No results returned from ESA query")
-    return(None,None)
-
-def getPageContents(url, verify):
-    hostname = urlparse(url).hostname
-    session = requests.Session()
-    session.mount(hostname, HTTPAdapter(max_retries=10))
-    page = session.get(url, timeout=60, verify=verify)
+    page = session.get(search_url, timeout=60, verify=verify)
```

Comment:
`page` is ok, but I almost always write `response = ...` for requests calls
should include `page.raise_for_status()` here